native/com_wolfssl_WolfSSL.c

```diff
@@ -1336,6 +1336,76 @@ JNIEXPORT jstring JNICALL Java_com_wolfssl_WolfSSL_getAvailableCipherSuitesIana
     return retString;
 }
 
+#ifdef WOLFSSLJNI_USE_NATIVE_CRYPTOCB
+/**
+ * Default native wolfSSL crypto callback function.
+ * This is called by default when wolfJSSE's WolfSSLProvider.registerDevId()
+ * is called, and is called by the native JNI API below.
+ *
+ * This function should be directly edited here to meet required
+ * functionality, or re-implemented and the registration point in the following
+ * function below should be edited:
+ *
+ * Java_com_wolfssl_WolfSSL_wc_1CryptoCb_1RegisterDevice()
+ */
+int DefaultNativeCryptoDevCb(int devId, wc_CryptoInfo* info, void* ctx)
+{
+    int ret = CRYPTOCB_UNAVAILABLE;
+    (void)devId;
+    (void)info;
+    (void)ctx;
+
+    /* Return CRYPTOCB_UNAVAILABLE to bypass HW and use SW. Edit function
+     * body here for your correct/expected functionality. */
+    return ret;
+}
+#endif /* WOLFSSLJNI_USE_NATIVE_CRYPTOCB */
+
+JNIEXPORT jint JNICALL Java_com_wolfssl_WolfSSL_wc_1CryptoCb_1RegisterDevice
+  (JNIEnv* jenv, jclass jcl, jint devId)
+{
+#ifdef WOLF_CRYPTO_CB
+    #ifdef WOLFSSLJNI_USE_NATIVE_CRYPTOCB
+        return wc_CryptoCb_RegisterDevice((int)devId,
+                    DefaultNativeCryptoDevCb, NULL);
+    #else /* Lookup the devId and see if it matches a known implementation */
+        #if defined(HAVE_CCBVAULTIC) && defined(WOLF_CRYPTO_CB_CMD)
+        #include "ccb_vaultic.h"
+        if(devId == CCBVAULTIC420_DEVID) {
+            return wc_CryptoCb_RegisterDevice((int)devId,
+                                              ccbVaultIc_CryptoDevCb, NULL);
```

Comment:
Why not register this in wolfCrypt_Init()?

Reply:
We'll track that as a "nice to have" for future work.  Need to reconsider leaving this outside of wolfSSL and what the correct model should be.  Good chat this morning.